crypto/trace.c

```diff
@@ -519,3 +519,26 @@ void OSSL_trace_end(int category, BIO * channel)
     }
 #endif
 }
+
+int OSSL_trace_string(BIO *out, int text, int full,
+                      const unsigned char *data, size_t size)
+{
+    unsigned char buf[OSSL_TRACE_STRING_MAX + 1];
+    int len, i;
+
+    if (!full && size > OSSL_TRACE_STRING_MAX) {
+        BIO_printf(out, "[len %zu limited to %d]: ",
+                   size, OSSL_TRACE_STRING_MAX);
+        len = OSSL_TRACE_STRING_MAX;
+    } else {
+        len = (int)size;
+    }
+    if (!text) { /* mask control characters */
+        for (i = 0; i < len; i++, data++)
+            buf[i] = (char)*data < ' ' ? ' ' : *data;
```

Comment:
127 is normally considered a control character too.
iscntrl() or ossl_iscntrl() perhaps?

Reply:
Good point - now using `ossl_iscntrl()`
and excempting the `\n` case in order to preserve existing newlines.